### ISSUE TITLE
Banned users cannot even read the forum

### ISSUE THREAD

**Issue by nickgammon** [arduino/forum-issues#177]
Once a user is banned, as I just found out, they don't even have read access. This is a bit harsh. If someone is a pest and gets a week's ban, surely they should still be able to browse the forum, without logging in. After all they can always go to a different PC and view forum posts.

**Comment by matteosuppo** [comment]
When you add a new ban you can choose wether the ban is a "Full Ban", denying access, or a "Partial Ban", denying posting, registering or login.

The moderators should decide which kind of ban to apply.

**Comment by nickgammon** [comment]
There is no choice about type of ban if you do a "delete user".

**Comment by nickgammon** [comment]
Also the automatic ban system denies users read access to the forum. If they are accidentally banned, as happened today, they can't even read the forum (for example to get help on how to get unbanned).

**Comment by fickleview** [comment]
For unknown reason, I have been banned on the Arduino forum?
I have never posted anything remotely abusive.
I have no other way to discover why.
Can you investigate Nick?

**Comment by fickleview** [comment]
My Arduino name is George Matthews.

**Comment by nickgammon** [comment]
Please state the exact message you get when you try to connect. I cannot see any ban against your username. Possibly your IP address is banned but I cannot see how to search IP address bans. I binary search of the IP bans pages (all 79 of them!) does not reveal your IP address as banned.

**Comment by fickleview** [comment]
On Thu, Dec 24, 2015 at 4:53 PM, Nick Gammon <notifications@github.com>
wrote:

> Please state the exact message you get when you try to connect. I cannot
> see any ban against your username. Possibly your IP address is banned but I
> cannot see how to search IP address bans. I binary search of the IP bans
> pages (all 79 of them!) does not reveal your IP address as banned.


​I have been using a VPN, so I tried login without VPN and have the same
error
​
"​
Sorry George Matthews, you are banned from using this forum!
This ban is not set to expire.
​"​

Current VPN IP 173.199.65.n

**Comment by nickgammon** [comment]
I cannot see any ban on George Matthews on the forum.

**Comment by fickleview** [comment]
On Fri, Dec 25, 2015 at 5:34 PM, Nick Gammon <notifications@github.com>
wrote:

> I cannot see any ban on George Matthews on the forum.


​Would you please push this issue upstream to a super admin?​



-- 

George Matthews *at Fickleview*
   44.310123, -64.410324


*2515 Hwy 331 West LaHaveRR#1 Pleasantville NS B0R 1G0*
Home 902-704-2479

**Comment by nickgammon** [comment]
I am not an admin (just a moderator) however your issue is being discussed amongst the moderators, and the forum administrators are copied into that mailing list. I'll send a direct request, however I warn you that in the middle of the Xmas holidays you may not get a fast response.

**Comment by fickleview** [comment]
On Fri, Dec 25, 2015 at 9:57 PM, Nick Gammon <notifications@github.com>
wrote:

> I'll send a direct request, however I warn you that in the middle of the
> Xmas holidays you may not get a fast response.


​That is perfectly reasonable... and thank you for your help!​



-- 

George Matthews *at Fickleview*
   44.310123, -64.410324


*2515 Hwy 331 West LaHaveRR#1 Pleasantville NS B0R 1G0*
Home 902-704-2479

**Comment by nickgammon** [comment]
I should also add that the forum has known bugs and problems. For example, finding an IP ban is extremely hard, as we have about 70 pages of IP addresses to look through to find a particular one. A binary search speeds it up a bit, but not a lot. I have passed your problem and request onto a senior forum administrator. You could try narrowing it down, eg. does another username work? A different IP address? (try Tor), or a user associated with a different email address?

**Comment by fickleview** [comment]
Update...
I am on Mac OS X 10.11.2
Safari web browser Version 9.0.2 (11601.3.9)
Will not work even for 'Guest'

Switch to Google Chrome Version 47.0.2526.106 (64-bit)
Works as expected!

As to why Mac Safari is denied is unknown at this point.

On Fri, Dec 25, 2015 at 10:12 PM, Nick Gammon <notifications@github.com>
wrote:

> I should also add that the forum has known bugs and problems. For example,
> finding an IP ban is extremely hard, as we have about 70 pages of IP
> addresses to look through to find a particular one. A binary search speeds
> it up a bit, but not a lot. I have passed your problem and request onto a
> senior forum administrator. You could try narrowing it down, eg. does
> another username work? A different IP address? (try Tor), or a user
> associated with a different email address?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/arduino/forum-issues/issues/177#issuecomment-167271389>
> .
>



-- 

George Matthews *at Fickleview*
   44.310123, -64.410324


*2515 Hwy 331 West LaHaveRR#1 Pleasantville NS B0R 1G0*
Home 902-704-2479

**Comment by Andy2No** [comment]
I've just tried to return to the forum to thank someone for helping me (I was too tired yesterday) and found I have been banned ("sock puppet"?) and that the ban is not due to expire.  I can't read any part of the forum, as Nick Gammon said at the start of this issue.

I hadn't used the forum in a long time.  I raised an issue with using the IDE (couldn't hide the console to make full use of the screen), received some help and discussed it briefly .  I wasn't rude, and I didn't post spam.

This leaves me with a very bad feeling.  I'm feeling let down, and offended, frankly.  I was trying to participate in the forum, and I expected to use it again.  Now, I can't even read any part of it.

Why does this happen?  Is it really an automatic process?  If so, why hasn't it been fixed, after all this time?  I can't think of any better way to drive people away from your forum, and create ill feeling towards your organisation, than letting people who are trying to take part be arbitrarily banned in this rather extreme way, without even an explanation.

**Comment by Andy2No** [comment]
Here is the exact error message I receive:

![arduino_forum_banned](https://cloud.githubusercontent.com/assets/8021302/25573713/de0c4fba-2e3f-11e7-9019-e1302913444a.gif)

**Comment by per1234** [comment]
I have the exact same message and I've been an active member of the forum for years without a single warning so I'm sure one of the moderators would have messaged me if there was an issue. There has been some work on the forum in the last few days and we have seen a lot of problems. Someone else had reported a false ban already. So don't jump to conclusions yet. If you've done nothing wrong I'm sure this was an error and we'll get you reinstated ASAP.

**Comment by Andy2No** [comment]
Thanks, per1234.

Personal feeling aside, my point really, is how this makes people feel when it happens to them, and therefore how serious an issue this is.  It needs to be fixed.

I've since done some more googling.  One guy was banned while trying to make his first post, shortly after registering.  Another found he could get around it by clearing cookies, site history and so on, but that it kept recurring.

I have managed to get back in, for now - and didn't even need to log in.  I cleared my recent history (cookies, site preferences etc).  People won't know to do that though.  I didn't.

Here are a couple of links.  I hope I don't get banned from Github for posting them too :)  The problem has apparently been going on for a long time, or has kept resurfacing:

https://github.com/arduino/forum-issues/issues/214

This was what solved it for me (for now), though I had to read it through Googles cached page, because I was banned from seeing it:

http://forum.arduino.cc/index.php?topic=211822.0

**Comment by Andy2No** [comment]
For the record, here's a link to the issue where you posted your own error message, pert, to tie them together:

https://github.com/arduino/forum-issues/issues/233

**Comment by nickgammon** [comment]
I've passed on your comments to the administrators. Sorry for any inconvenience. I personally don't have any control over this stuff.

**Comment by Andy2No** [comment]
Thanks, Nick.

**Comment by feelgoodgirl** [comment]
Heyy,

I am banned aswell. I dont really dont know why. my user name is FeelGoodGirl. I got the same message as Andy2No.

I really liked to use the arduino forum again... can you please help me?

greets,
Yvette

**Comment by paulalting** [comment]
Hello Nick, I too have just been banned, so it seems.
I was able to make a replay post at 21h24 EAST fine and then some 30 minutes later while researching more on Wiznet 5500 Ethernet I get a message the same  as Andy2No and Yvette get, and am in shock as to how this can happen.

Nick, my last post, as I have access to the link in one on my browser tabs is [http://forum.arduino.cc/index.php?topic=474152.msg3242798#new](url).

As there is no way for me to contact anyone through the forum to ask why, I would really appreciate it if you were able to check on the situation for me.

Kind regards,
Paul (From Tasmania)

**Comment by chlowden** [comment]
Hello
I have just been blocked out with the same message.
I exited Firefox and logged back into the site with Chrome with the same login etc with no problem at all
Very annoying.
I cleared out the cache from Firefox but to no avail. Any thoughts what cookies etc might be being used?
Thanks

**Comment by guarrod** [comment]
I think I was banned for leaving the session open :( can you help me? My username is Guarrod

**Comment by paulalting** [comment]
Right, I then went into Chrome and was able to login fine on the same machine, Linux Mint LMDE 2, so that was good.

Next, I went back to Firefox and went into preferences and selected Privacy and then selected 'remove individual cookies' and from the list I had two that were 'forum.arduino.cc', which I removed.

Then back to an open tab that was already loaded at the Arduino forum with the 'Bad Sock Puppet' message and simply navigated back a page and viola, le page opens good again.

So, at this stage, all seems to be good and nice in the world of Arduino.
Paul.

**Comment by matteosuppo** [comment]
Seems like there is an automatic ban trigger that's too sensitive. I think I found it and disabled it. Please let us know if you are still having trouble with the forum

**Comment by chlowden** [comment]
Erased about 30 arduino cookies and now i can login as before. Thanks

**Comment by Andy2No** [comment]
Could you also change the terms of the ban, matteosuppo?

The original point of this thread (before my bug report) was that banned users are banned from reading the forum, not just posting in it.  

Banning people from posting makes good sense, if there's a valid reason for banning them, but banning people from reading it is just bizarre.  It also makes it harder for people to find a solution to the problem of having been banned arbitrarily - they can't find the answers they need in the forum.

**Comment by Andy2No** [comment]
Yvette, Guarrod and chlowden,

The actual step I took in Waterfox (same thing as Firefox, pretty much) was to go to History, then Clear Recent History.  I had Cookies, Cache, Offline Website Data and Site Preferences all ticked, and no others, and the "Time range to clear" set to Today.

Possibly not all of that was needed, and it remains to be seen if it's only a temporary fix, but I was able to read the forum and post afterwards.

**Comment by BallscrewBob** [comment]
Doesnt seem to affect IE11 users quite as much as I was able to get back in using it.
However it remained an issue on chrome for a while despite doing all of the above steps.
AM back in with chrome at the moment but am unsure how long it will last.

**Comment by Shingon7** [comment]
Good Morning,

My name is Alan. I also got banned from the arduino forums last night. It displayed the following message:

Sorry Alan777, you are banned from using this forum!
Sock puppet.
This ban is not set to expire.

Could any of you help me see why I got banned and how may I become unbanned? I'm currently working on an arduino project and need access to the forums. Your help is greatly appreciated! :)

**Comment by Andy2No** [comment]
Alan,

Read this thread, or at least the last couple of posts.

Why?  Probably the same bug.
How?  See above.

**Comment by guarrod** [comment]
Wow @chlowden You're right!

**Comment by Shingon7** [comment]
Andy2No, thank you so much for your reply! Sorry for not reading the whole thread, I'm currently at work and skipped a few messages, but I will follow these recommendations as soon as I get home and will let you guys know how it goes! Again, thanks for the reply and hope this takes care of the issue! Have a good one! :)

**Comment by nickgammon** [comment]
@matteosuppo - it still doesn't make any sense to not allow banned people to **read** the forum. What harm can that do? If they can't read the forum then there is no point in us making posts telling them how to fix things if they are accidentally banned.

**Comment by nickgammon** [comment]
Unfortunately the ban list (and ban log) is not searchable. There are hundreds of pages of events that we can see (the first 10 seem to be the same person/bot trying every 30 seconds for hours). It just isn't feasible to read up to page 325 to see if your name appears somewhere. I've asked them to change this.

Meanwhile, try a different browser and try deleting your Internet cache, and cookies, at least for arduino.cc site.

**Comment by feelgoodgirl** [comment]
its not fun to read that so many persons havind the same banning problems. 

i tryed this morning again. and i can use it again!! i am so happy about it, because i use it allot with all projects.

greets, yvette

**Comment by per1234** [comment]
@nickgammon are you satisfied with @Serjlee's resolution of this issue (from https://github.com/arduino/forum-issues/issues/248#issuecomment-309401421):
>All previous and future bans now always allow read-only access to the forum.

If so, I think this issue could be closed.

**Comment by RCAddict84** [comment]
Im absolutely disgusted with the Arduino forums and DO NOT RECOMMEND them to anyone especially minors due to the the abuse from members and Admins. 

Im new to programing and posted up a project in the help section for some help,  I got slammed and insulted for making a mistake in my code by a member called PaulS

so as a response I explained this is a help forum and if people knew how to code with out making mistakes they wouldn't be asking for help and edited my post to not include the code as i no longer wished to share it, and after all it is mine, I spent ages making it.
And finally I reported the issue to Admin.

the next morning i log on to find ive been insulted via Admin and banned.... for vandalising the forum?? where in the rules does it say i cant edit MY posts?

HOW IS THAT FAIR!!!!!!! i cant even message a higher up admin to make a complaint, and since making complaints like this one, my thread has been edited further to hide how admin dealt with it

if u have admin privileges, have a look at the thread  and undelete the posts via Admin(coding badly) and PaulS

and tell me if u think that fair and with in the Arduinos code of conduct.
basically i want to make a complaint with the the ACCC, but ive been informed in need to try complaining to Arduino first

https://forum.arduino.cc/index.php?topic=524610.msg3577708#msg3577708

**Comment by RCAddict84** [comment]
all I want is the threads I created deleted from the forums. I no longer wish to share my material and I dont want to have my user name or email associated with Arduino. and looking through the forum rules there appears to be noting ive agreed to that would prevent that

here are the 2 threads i want removed forever or,,, at least put back to how they were so others can have a read and make there own decision is they want to join the forum

https://forum.arduino.cc/index.php?topic=521326.0

https://forum.arduino.cc/index.php?topic=524610.msg3577708#msg3577708

**Comment by Andy2No** [comment]
That's not the sort of banning problem this topic was about, RCAddict84, but I do know what you mean.

There are one or two people (present company definitely excepted) who seem to have nothing better to do than hang out in that forum being self important, and often quite hostile.  It makes me wary of going there.

FWIW, I had a post deleted where I responded to someone calling me a whiner, for reporting a bug in the IDE (bug, or unfortunate user interface design, take your pick).  I think I remained polite, but I couldn't swear to it.  His post was deleted too.  

That is just the way their admins do things, I guess.  I didn't take offence at that, but I can understand the frustration when that sort of editing happens.  If I'd been banned too, it would definitely have made me angry.

As for new members realising what the forum is like - it didn't take me long to realise I don't enjoy being there much, hence my low post count over a few years of membership.

We really are off topic though, so I'd suggest starting a new thread.  Github is about software development though, so I don't think this is really the place.  It's a good place to point out bugs in the forum software, but it's not really about how the admins of the forum do their thing, or how some of the members behave.

It's a shame the main Arduino forum is as hostile as it is - enough of the time to matter, IMHO.  I don't think it's limited to hostility to beginners either.  I don't regard myself as a beginner to microcontrollers, or to electronics, though I'm certainly no expert.  Unfortunately, it hosts some massive but egg shell fragile egos.  

I won't suggest you put up with it, because I don't believe we should have to.  All we can really do is vote with our feet.  I suspect a lot of people have done that - it's a very popular hobby but a remarkably small forum, considering.

Whether it's worth appealing against an actual ban, if that's happened, is a matter for the individual.  Personally, I think I'd give up, at that point - I don't use the forum much, I rarely enjoy it or benefit from it when I try (see above) and I could live without it.

As for having your content deleted, I agree, you ought to have the right, but I guess it comes down to the forum Terms and Conditions - which almost no one has ever read, I expect.   Even so, your posts are interesting, and a contribution to the hobby.  A couple of years from now, people would see your post count stopped shortly after you joined, and get the message that you weren't happy.

I've seen that plenty of times, and that was my conclusion.  I've since been in touch with one person I'd noticed, who made two very interesting posts, then no more - in the course of buying an FM synthesis module he made, and once tried to discuss on the forum...

I thought there was another Arduino forum, but I can't find it at the moment.  If you buy from Sparkfun, there's their own user forum to try.

Returning to something a bit closer to the topic - have you checked again to make sure you really are banned?  Bugs in forums can resurface sometimes, because the code base changes quite a lot.

The reason this thread got started was because "apparent" banning was happening often, due to a bug - people were told they'd been banned, when they hadn't.  At the time, being banned meant you couldn't read the forum, though I understand that part of the problem has been dealt with.

Things you can try include clearing your browser cookies, etc, or trying a different browser, especially if you've never used it for that forum before.

If you've been specifically told you were banned, this won't help, of course.

**Comment by nickgammon** [comment]
The forum is quite flaky in parts, for example [I can't change my screen name](https://github.com/arduino/forum-issues/issues/234) which has been sitting around as a bug report for almost a year with no-one assigned to it.

Also the pagination of post subjects is out by a couple of lines which is confusing.

---

>  ive been insulted via Admin and banned.... for vandalising the forum?? where in the rules does it say i cant edit MY posts?

It can be very frustrating on a forum (or Stack Exchange) where someone makes a lengthy post, you make a big effort to reply, and then they vandalize the post to read "asdfasdf" which means the replies now make no sense.

I must admit I can't find the terms and conditions for the Arduino Forum, if indeed they exist. However generally you are expected to realize that your posts are released into the public domain, and that attempts to change the entire meaning of a post are likely to be resisted, as a courtesy to other users.

> I thought there was another Arduino forum, but I can't find it at the moment.

There is [Arduino Stack Exchange](https://arduino.stackexchange.com/) which is **not** a forum - it is a Q&A format. You ask a question and people attempt to reply. However there is quite a bit of crossover between that and forums. The Stack Exchange sites are a bit different to forums too, in that they are generally "community moderated". If a post gets closed it will generally only be because at least 5 other users (with sufficient reputation) voted to close it.

Also posts there can be edited by people of sufficient reputation. Reputation is gained by asking and answering questions, and having your answers voted up (amongst other things).

*Disclaimer* - I am a moderator at Arduino Stack Exchange, however the workload there is pretty light as the community moderation does 90% of the work. Moderators there are only required for extreme situations, or cases where some sort of manual intervention is required.

---

> Ive been informed in need to try complaining to Arduino first

I suggest you send an email to **webmaster@arduino.cc** - I believe they get read, because occasionally the admins contact us (moderators) to ask why we did something.

However my advice is to let it go. Your post(s) will quickly drop off into page 50 or somewhere where no-one will notice them.

It's a pity that the Arduino Forum doesn't retain an edit history, so claims about "he said/she said" are hard to verify. On Stack Exchange all edits are retained, so you can quickly see if the current post is different from what was originally asked.

**Comment by RCAddict84** [comment]
People need to understand that in a public domain, the Original poster owns and controls his or her content of the post. No one has the right to prevent them from deleting or editing there own comments.

if someone makes a lengthy reply post and The OP deletes or edits there post that’s just the risk u take.


I cant just let this go.... after complaining on loads of other forums I’m getting feedback that this is a big issue with the Arduino forms and there is active cliques that look down and pray on new people


Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by RCAddict84** [comment]
Thankyou for the reply, I feel a lot better after reading that. Hopefully I can get my mojo back and start programing again, I haven’t touched anything since. Its left a big negative void

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by BallscrewBob** [comment]
Can this be closed and LOCKED ?
Clearly it has become a "rant" and of no technical use.

**Comment by nickgammon** [comment]
> From: Nick Gammon<mailto:notifications@github.com>

@ RCAddict84 - can you please not **quote** an email which has the exact content as in the post above you here? It is confusing to read something that is "from you" which is actually what I said directly above.

Ditto for  Andy2No's comment.

After what has been said I wouldn't dare to do it myself (and probably don't have the rights to anyway) but you might consider editing your responses here and just sticking to the point **you** are making. Sure, if you were quoting from some other place completely that would make more sense.

---

> there is active cliques that look down and pray on new people

I really don't think that is true. However there are some people who get fed up with questions that are basically "do my homework for me" or "write my code for me" or "answer this question because I am too lazy to use Google". I'm not accusing you of these things, I'm just pointing out that some of the frequent posters have to put up with this every single day. Sometimes they can get a little terse because of it. :)

**Comment by RCAddict84** [comment]
haha sorry mate... when replying to emails I generally reply at the top while leaving there message, I thought its a highly common practice and recommend by communication experts. Say for example i got your reply but had no context to what it was... it took me a bit to realise it was in response to this thread

Maybe its just forums???? I jump on here and already some random dude wants to silence me... all i think is why? cant I have my say? is not ranting/venting a good thing? how is this effecting him???

I just dont understand? why lock or close the thread???? if u dont like it keep scrolling, its that simple.

Personally i think the problem is people have no respect or honour and act different behind a computer screen

**Comment by per1234** [comment]
This is a bug tracker. It's a place to report problems with the Arduino forum software. This is **NOT** an appropriate place to rant or vent. The people who are subscribed to this repository get an email every time someone comments on an issue thread. We subscribe because we want to be notified about these issues. Off topic conversations such the one you are perpetuating result in a lot of spam. You have now caused 150 emails to be sent about something that is absolutely not a bug in the Arduino Forum software and isn't even related to the original topic of this issue. GitHub is the place we come to work on open source projects, not chat.

An appropriate place to discuss your situation is the Arduino forum. I realize you were banned but, unless there are some additional non-public actions I can't see, I'm certain it was only a short temporary ban that has long since expired. If you will only use the forum in a mature and respectful manner and not be so quick to take offense then we will be happy to help you with your project. If you're not capable of doing so just accept that you're not a good fit for the Arduino community and move on.

**Comment by per1234** [comment]
Closing as resolved by https://github.com/arduino/forum-issues/issues/248#issuecomment-309401421
